nodejs/lancedb/arrow.ts

```diff
@@ -103,25 +103,12 @@ export type IntoVector =
   | number[]
   | Promise<Float32Array | Float64Array | number[]>;
 
-export type FloatLike =
-  | import("apache-arrow-13").Float
-  | import("apache-arrow-14").Float
-  | import("apache-arrow-15").Float
-  | import("apache-arrow-16").Float
-  | import("apache-arrow-17").Float;
-export type DataTypeLike =
-  | import("apache-arrow-13").DataType
-  | import("apache-arrow-14").DataType
-  | import("apache-arrow-15").DataType
-  | import("apache-arrow-16").DataType
-  | import("apache-arrow-17").DataType;
-
```

Comment:
These cannot appear in our public API without forcing users to install all versions.

Reply:
export function isDataType(value: unknown): value is DataType {
  return sanitizeType(value) instanceof DataType;
}

Reply:
(actually, I don't think we even need the `instanceof`, `sanitizeType` will throw if the input isn't type-like.

Reply:
Or, (sorry for stream of consciousness), we should replace all calls to `isDataType` with `sanitizeType`.

Reply:
No worries, I find this Typescript dependency resolution stuff very confusing so I appreciate the help.